File: Sorting/Bubble Sort/bubbleSort.cpp
// Program to sort the array using "Bubble Sort"

#include<iostream>
#define maxSize 100

using namespace std;

// Function to sort the array using Bubble Sort
void bubbleSort(int array[], int size)
{
    int temp;
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            if(array[j] > array[j+1])
            {
                temp = array[j];
                array[j] = array[j+1];
                array[j+1] = temp;
            }
        }
    }

    // Printing the sorted array
    cout << "\nThe sorted array: " << endl;
    for(int i=0; i<size; i++)
    {
        cout << array[i] << " ";
    }
}

int main()
{
    int array[maxSize], size;

    // Scanning the size of the array
    cout << "Enter the size of the array: ";
    cin >> size;

    // Scanning the elements of array
    cout << "Enter the " << size << " elements of array: " << endl;
    for(int i=0; i<size; i++)
    {
        cin >> array[i];
    }

    // Calling bubbleSort function
    bubbleSort(array,size);

    return 0;
}

File: Sorting/Bubble Sort/bubble_sort.py
# Program to perform bubble sort

# Function to scan the list
def scan_list(list_1, size):
    print(f"Enter the {size} elements of the array:")
    for i in range(size):
        element = int(input(f"Element [{i}]: "))
        list_1.append(element)


# Function to print the list
def print_list(list_1):
    print(list_1)


# Function to perform bubble sort
def bubble_sort(list_1, size):
    for i in range(size-1):
        for j in range(size-1-i):
            if list_1[j] > list_1[j+1]:
                (list_1[j], list_1[j+1]) = (list_1[j+1], list_1[j])


# Driver Function
if __name__ == '__main__':
    list_1 = []
    size = int(input("Enter the number of elements of the list: "))
    scan_list(list_1, size)
    print_list(f"List: {list_1}")
    bubble_sort(list_1, size)
    print_list(f"Sorted List: {list_1}")
